File: app/src/main/java/com/example/product1/TovarDetails.java
package com.example.product1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class TovarDetails extends AppCompatActivity {
    TextView name_product;
    TextView price;
    TextView count;
    TextView manufacturer;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tovar_details);

        name_product = findViewById(R.id.name_product);
        price = findViewById(R.id.price);
        count = findViewById(R.id.count);
        manufacturer = findViewById(R.id.manufacturer);
    }


    public void back(View view){
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/example/product1/TovarDetails.kt
package com.example.product1;

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.LifecycleObserver
import timber.log.Timber


class TovarDetails: AppCompatActivity(), LifecycleObserver {
    var name_product: TextView? = null;
    var price : TextView? = null;
    var count: TextView? = null
    var manufacturer: TextView? = null

    private lateinit var dessertTimer: DessertTimer

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState);
        Timber.i("onCreate Called");
        dessertTimer = DessertTimer(this.lifecycle)

        setContentView(R.layout.activity_tovar_details);

        name_product = findViewById(R.id.name_product);
        price = findViewById(R.id.price);
        count = findViewById(R.id.count);
        manufacturer = findViewById(R.id.manufacturer);

        if (savedInstanceState != null) {
            dessertTimer.secondsCount = savedInstanceState.getInt(KEY_TIMER_SECONDS, 0)
        }

    }

    fun back(view: View) {
        intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        outState.putInt(KEY_TIMER_SECONDS, dessertTimer.secondsCount);
        Timber.i("onSaveInstanceState Called");
        super.onSaveInstanceState(outState);
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState);
        Timber.i("onRestoreInstanceState Called");
    }
    /** Lifecycle Methods **/
    override fun onStart() {
        super.onStart();
        Timber.i("onStart Called");
    }

    override fun onResume() {
        super.onResume();
        Timber.i("onResume Called");
    }

    override fun onPause() {
        super.onPause();
        Timber.i("onPause Called");
    }

    override fun onStop() {
        super.onStop();
        Timber.i("onStop Called");
    }

    override fun onDestroy() {
        super.onDestroy();
        Timber.i("onDestroy Called");
    }

    override fun onRestart() {
        super.onRestart();
        Timber.i("onRestart Called");
    }
}